core_java/twitter/src/main/java/ca/jrvs/apps/twitter/controller/TwitterController.java

```diff
@@ -0,0 +1,74 @@
+package ca.jrvs.apps.twitter.controller;
+
+import ca.jrvs.apps.twitter.model.Coordinates;
+import ca.jrvs.apps.twitter.model.Tweet;
+import ca.jrvs.apps.twitter.service.Service;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@org.springframework.stereotype.Controller
+public class TwitterController implements Controller {
+
+  private static final String COORDINATE_SEPARATOR = ":";
+  private static final String COMMA = ",";
+
+  /**
+   * Dependency
+   */
+  private Service service;
+
+  /**
+   * Dependency setup in constructor
+   *
+   * @param service
+   */
+  @Autowired
+  public TwitterController(Service service) {
+    this.service = service;
+  }
+
+  /**
+   * Parse user argument and post a tweet by calling service classes
+   *
+   * @param args
+   * @return a posted tweet
+   * @throws IllegalArgumentException if args are invalid
+   */
+  @Override
+  public Tweet postTweet(String[] args) {
+    String status = args[1];
+    String[] location = args[2].split(COORDINATE_SEPARATOR);
+    float longitude = Float.parseFloat(location[0]);
+    float latitude = Float.parseFloat(location[1]);
+    float[] longLat = {longitude, latitude};
+    Coordinates coordinates = new Coordinates(longLat, "Point");
```

Comment:
you could write a util static method which takes two `long` arguments and return a new coordinate. I saw you repeat this code over and over in JUint. Remember DRY. (this fix is optional)